File: examples/node/getting-started.js
globalThis.WebSocket = require("websocket").w3cwebsocket;

const kaspa = require("../kaspa-wasm32-sdk/nodejs/kaspa");
const { RpcClient, Resolver, Encoding } = kaspa;

const networkId = "mainnet";
const rpcUrl = undefined;

(async () => {
  const rpc = new RpcClient({
    networkId,
    encoding: Encoding.Borsh,
    ...(rpcUrl ? { url: rpcUrl } : { resolver: new Resolver() }),
  });

  try {
    await rpc.connect();

    const dagInfo = await rpc.getBlockDagInfo();
    console.log(dagInfo);
    console.log("Pruning point:", dagInfo.pruningPointHash);
  } finally {
    await rpc.disconnect().catch(() => {});
  }
})().catch((error) => {
  console.error(error);
  process.exit(1);
});

File: examples/rust/src/bin/getting_started.rs
use std::sync::Arc;
use std::time::Duration;

use kaspa_rpc_core::api::rpc::RpcApi;
use kaspa_wrpc_client::{
    client::{ConnectOptions, ConnectStrategy},
    prelude::{NetworkId, NetworkType},
    KaspaRpcClient, Resolver, WrpcEncoding,
};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let network_id = NetworkId::new(NetworkType::Mainnet);
    let rpc_url: Option<&str> = None;

    let client = Arc::new(KaspaRpcClient::new_with_args(
        WrpcEncoding::Borsh,
        rpc_url,
        if rpc_url.is_some() {
            None
        } else {
            Some(Resolver::default())
        },
        Some(network_id),
        None,
    )?);

    let options = ConnectOptions {
        block_async_connect: true,
        connect_timeout: Some(Duration::from_millis(5_000)),
        strategy: ConnectStrategy::Fallback,
        ..Default::default()
    };

    client.connect(Some(options)).await?;

    let dag_info = client.get_block_dag_info().await?;
    println!("{dag_info:#?}");
    println!("Pruning point: {}", dag_info.pruning_point_hash);

    client.disconnect().await?;

    Ok(())
}
